### PR TITLE
Add support for webhook ca bundle

### PR DESCRIPTION
**Pull request by thedebugger**
Hi, this PR adds support to set caBundle for webhooks explicitly. This is useful for following reasons:
* It is needed when pilot is configured with a well known certificate
* It improves istio gitops installation because pilot don't need to patch webhook caBundle and failurePolicy which conflicts with git config.
* Probably makes installing webhooks easier during remote istio installation?

### CODE REVIEW

File: manifests/charts/istio-control/istio-discovery/values.yaml
Comment (hanxiaop) [inline manifests/charts/istio-control/istio-discovery/values.yaml:495]: I think this won't work, the webhook caBundle will be patched by Istiod's certs.

Comment (thedebugger) [reply manifests/charts/istio-control/istio-discovery/values.yaml:495]: ohh this works fine if you disable the istio patching with env variable. We have been running Istio this way but we had to patch using kustomize to support override caBundle. This PR will help us get rid of kustomize patches

Comment (thedebugger) [inline manifests/charts/istio-control/istio-discovery/values.yaml:495]: any concerns with keeping value at the root? if yes, please suggest where to add this property. Reason I added here is it applies to both all webhooks (for now)

Comment (thedebugger) [reply manifests/charts/istio-control/istio-discovery/values.yaml:495]: For context, we have been running Istio with well known cert and Google CAS for workload certificate

**Review by howardjohn — COMMENTED** [review]
Overall LGTM

File: manifests/charts/istio-control/istio-discovery/values.yaml
Comment (hanxiaop) [reply manifests/charts/istio-control/istio-discovery/values.yaml]: > if you disable the istio patching with env variable.

This makes sense to me, thanks

**Review by hanxiaop — COMMENTED** [review]
You need to add a releasenote and make changes to https://github.com/istio/istio/blob/master/operator/pkg/apis/istio/v1alpha1/values_types.proto as well.

File: Makefile
Comment (thedebugger) [inline Makefile:22]: support override

Comment (howardjohn) [reply Makefile:22]: Its not clear why we need an override, but regardless
1. Please make this in a separate PR
2. These files are synced from https://github.com/istio/common-files/tree/master/files, so they will need to be changed there to have effect (same with the run.sh change)

**Review by howardjohn — COMMENTED** [review]
LGTM besides my comment

File: Makefile
Comment (thedebugger) [reply Makefile:22]: I'm on nixos and paths are different. I'll just back out these changes for now and open up new PR if needed

**Comment by thedebugger** [comment]
@howardjohn i removed the bashc changes. any other tests you want me to run before I merge this PR?

File: manifests/charts/base/templates/default.yaml
Comment (hzxuzhonghu) [inline manifests/charts/base/templates/default.yaml:27]: In some case we donot know CABundle before we run istiod up

**Review by hzxuzhonghu — COMMENTED** [review]
should stop reconcile webhookConfiguration in this case

**Review by howardjohn — APPROVED** [review]

File: manifests/charts/base/templates/default.yaml
Comment (thedebugger) [reply manifests/charts/base/templates/default.yaml:27]: right which is why this is a opt in - folks can set this value if they know ahead of time

**Review by craigbox — APPROVED** [review]
I think you need approval for docs to continue.  Let's see if this does it

👍

**Review by costinm — APPROVED** [review]
Great idea. Approved - but few cosmetic comments:

- name is too generic, validationCA for what ? I would use webhookValidationCA
- perhaps adding directly under values is better - 'base' is the name of the chart, no need to have it in the values ( all values apply to the base chart ).

**Comment by craigbox** [comment]
🎉

**Comment by craigbox** [comment]
🎉

**Pull request merged**
